qatrfm/domain.py

```diff
@@ -160,7 +165,9 @@ def wait_for_qemu_agent_ready(self, timeout=300):
             except libutils.TrfmCommandFailed:
                 i += 1
                 time.sleep(10)
-        raise libutils.TrfmDomainTimeout
+        self.logger.warning("Qemu-agent is not available on the domain {}."
+                            .format(self.name))
+        self.qemu_agent = False
```

Comment:
I would raise that exception as well. So this method acts always the same first and all others.

```
raise libutils.TrfmQemuAgentNotReady("Qemu-agent is not running on the domain")
```

I'm not sure if we should try qemu_agent always or only the first time. Is there a possibility that qemu_agent can be started afterwards?

Reply:
The purpose of the PR is not to exit if qemu-agent is not running in the VM, which might be the case for some images (e.g. non-openqa images)

Reply:
> I'm not sure if we should try qemu_agent always or only the first time. Is there a possibility that qemu_agent can be started afterwards?

It is actually possible, just checked, but I don't think it's desired and users will really do that, but it's possible

    systemctl start qemu-ga@virtio\\\\x2dports-org.qemu.guest_agent.0.service

Reply:
> The purpose of the PR is not to exit if qemu-agent is not running in the VM, which might be the case for some images (e.g. non-openqa images)

but on next call to execute_cmd we will exit anyway , right ? So do you want to exit if you not able to execute_cmd or not ?
Also important point to call things what they really are. And I have some doubts that only reason why you can not execute command is the fact that agent is not running. The only thing which you know for sure when your command timeout is that command timeout :)

Reply:
>  So do you want to exit if you not able to execute_cmd or not ?

I don't want to exit at deployment time (= qemu-agent is not mandatory for an image). But in the test, that should be known and the user who writes the test will know if execute_cmd can be used or not.